packages/flutter/lib/src/rendering/sliver_multi_box_adaptor.dart

```diff
@@ -214,6 +219,27 @@ abstract class RenderSliverMultiBoxAdaptor extends RenderSliver
   /// The nodes being kept alive despite not being visible.
   final Map<int, RenderBox> _keepAliveBucket = <int, RenderBox>{};
 
+  List<RenderBox> _debugDanglingKeepAlives;
+
+  /// Indicates whether child order verification is enabled
+  ///
+  /// It will perform integrity check every time it is set
+  /// to true.
+  bool get debugChildIntegrityEnabled => _debugChildIntegrityEnabled;
+  bool _debugChildIntegrityEnabled = true;
+  set debugChildIntegrityEnabled(bool enabled) {
+    assert(enabled != null);
+    assert(() {
+      _debugChildIntegrityEnabled = enabled;
+      _debugVerifyChildOrder();
+      if (_debugChildIntegrityEnabled) {
+        assert(!_debugDanglingKeepAlives.isNotEmpty);
```

Comment:
I figure this assert in assert is more informative than return false

Reply:
you could also replace the entire `if` with `return !_debugChildIntegrityEnabled || _debugDanglingKeepAlives.isNotEmpty`

certainly in any case you don't want both `!` and `isNotEmpty`.

Reply:
if you do the `bool` thing below, this would become:

`return _debugVerifyChildOrder() && (!_debugChildIntegrityEnabled || _debugDanglingKeepAlives.isEmpty);`